### PR TITLE
Deprecate --no-print-logs option

### PR DESCRIPTION
**Pull request by vinaycalastry**
- Deprecated the --no-print-logs option.
- Added 2 unit tests, for ini and command line.
- Added `6328.deprecation.rst` file.
- Updated the `deprecations.rst` doc.

### CODE REVIEW

File: changelog/3238.deprecation.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Option ``--no-print-logs`` is deprecated and meant to be removed in a future release. If you use ``--no-print-logs``, please try out ``--show-capture`` and
+provide feedback.
+
+``--show-capture`` command-line option was added in ``pytest 3.5.0` and allows to specify how to
```
Comment (nicoddemus) [inline changelog/3238.deprecation.rst]:
```suggestion
``--show-capture`` command-line option was added in ``pytest 3.5.0`` and allows to specify how to
```

Comment (nicoddemus) [reply changelog/3238.deprecation.rst]: Great entry!

File: doc/en/deprecations.rst [hunk 2]
```diff
@@ -19,6 +19,17 @@ Below is a complete list of all pytest features which are considered deprecated.
 :class:`_pytest.warning_types.PytestWarning` or subclasses, which can be filtered using
 :ref:`standard warning filters <warnings>`.
 
+No Print logs (``--no-print-logs)
```
Comment (nicoddemus) [inline doc/en/deprecations.rst]:
```suggestion
``--no-print-logs`` command-line option
```

File: doc/en/deprecations.rst [hunk 3]
```diff
@@ -19,6 +19,17 @@ Below is a complete list of all pytest features which are considered deprecated.
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (nicoddemus) [inline doc/en/deprecations.rst]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/en/deprecations.rst [hunk 4]
```diff
@@ -19,6 +19,17 @@ Below is a complete list of all pytest features which are considered deprecated.
...
+
+.. deprecated:: 5.0
```
Comment (nicoddemus) [inline doc/en/deprecations.rst]:
```suggestion
.. deprecated:: 5.4
```

File: testing/deprecated_test.py [hunk 5]
```diff
@@ -44,3 +44,42 @@ def test_external_plugins_integrated(testdir, plugin):
 
     with pytest.warns(pytest.PytestConfigWarning):
         testdir.parseconfig("-p", plugin)
+
+
+def test_noprintlogs_is_deprecated(testdir):
```
Comment (nicoddemus) [inline testing/deprecated_test.py]: You can use parametrize and have a single test here:

```python
@pytest.mark.filterwarnings("default")
@pytest.mark.parametrize('config_mode', ["ini", "cmdline"])
def test_noprintlogs_is_deprecated_ini(testdir, config_mode):
    if config_mode == "ini":
        args = ()
        testdir.makeini(
            """
            [pytest]
            log_print=False
            """
        )
    else:
        assert config_mode == "cmdline"
        args = ("--no-print-logs",)

    testdir.makepyfile(
        """
        def test_foo():
            pass
        """
    )

    result = testdir.runpytest(*args)
    result.stdout.fnmatch_lines(
        [
            "*--no-print-logs is deprecated and scheduled for removal in pytest 6.0*",
            "*Please use --show-capture instead.*",
        ]
    )
```

File: changelog/3238.deprecation.rst (hunk 1 above)
Comment (vinaycalastry) [reply changelog/3238.deprecation.rst]: Fixed

File: doc/en/deprecations.rst (hunk 2 above)
Comment (vinaycalastry) [reply doc/en/deprecations.rst]: Changed this entry

File: doc/en/deprecations.rst (hunk 3 above)
Comment (vinaycalastry) [reply doc/en/deprecations.rst]: Fixed this

File: doc/en/deprecations.rst (hunk 4 above)
Comment (vinaycalastry) [reply doc/en/deprecations.rst]: Changed version to 5.4

File: testing/deprecated_test.py (hunk 5 above)
Comment (vinaycalastry) [reply testing/deprecated_test.py]: Thank you! I changed the unit test to use parametrized values now.

**Pull request closed without merging**